### PR TITLE
Use upstream containerd

### LINKED ISSUE #41: Forked containerd/containerd dependency
This library is using a forked version of `containerd/containerd`, and while upstream is at version 1.26, this is at 1.21, and I have a couple of questions:

- is there a plan for the changes in the fork to be pushed upstream?
- if that is taking longer, is there a plan to keep the branch used here up-to-date?

Thanks!

### PR DESCRIPTION
**Pull request by rumpl**
Updated the containerd dependency to user containerd/containerd since
the latest version can push to multiple repos.

For mount on push to work we are now adding the
"containerd.io/distribution.source" annotation before pushing.

Resolves #41

### CODE REVIEW

File: remotes/mount.go [hunk 1]
```diff
@@ -10,28 +10,38 @@ import (
 	"github.com/containerd/containerd/errdefs"
 	"github.com/containerd/containerd/images"
 	"github.com/containerd/containerd/remotes"
+	"github.com/docker/distribution/reference"
 	ocischemav1 "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
 )
 
-func newDescriptorCopier(ctx context.Context, resolver remotes.Resolver, sourceFetcher remotes.Fetcher, targetRepo string, eventNotifier eventNotifier) (*descriptorCopier, error) {
+var (
```
Comment (silvin-lubecki) [inline remotes/mount.go]: nit: should be a const

File: remotes/mount.go [hunk 2]
```diff
@@ -10,28 +10,38 @@ import (
...
+	// labelDistributionSource describes the source blob comes from.
```
Comment (silvin-lubecki) [inline remotes/mount.go]: Add a comment to explain where it was copied from?

File: remotes/mount_test.go [hunk 3]
```diff
@@ -40,3 +47,48 @@ func TestRemoteReaderAtShortReads(t *testing.T) {
 	assert.Equal(t, n, len(helloWorld))
 	assert.DeepEqual(t, helloWorld, actual)
 }
+
+// This test does that the `Handle` method does with the Annotations before
```
Comment (silvin-lubecki) [inline remotes/mount_test.go]: typo: that -> what?

File: remotes/fixup.go [hunk 4]
```diff
@@ -110,23 +110,21 @@ type FixupOption func(*fixupConfig) error
 
 // ResolverConfig represents a resolver and its associated OriginProviderWrapper
 type ResolverConfig struct {
-	Resolver              remotes.Resolver
-	OriginProviderWrapper OriginProviderWrapper
+	Resolver remotes.Resolver
```
Comment (jcsirot) [inline remotes/fixup.go]: Is that type still required since it only contains a resolver?

Comment (rumpl) [reply remotes/fixup.go]: Not really actually, I'll remove it

**Pull request merged**